src/AppInstallerCLICore/Commands/SettingsCommand.cpp

```diff
@@ -110,4 +111,104 @@ namespace AppInstaller::CLI
         context <<
             Workflow::ExportSettings;
     }
+
+    std::vector<Argument> SettingsSetCommand::GetArguments() const
+    {
+        return {
+            Argument { Execution::Args::Type::SettingName, Resource::String::SettingNameArgumentDescription, ArgumentType::Positional, true },
+            Argument { Execution::Args::Type::SettingValue, Resource::String::SettingValueArgumentDescription, ArgumentType::Positional, true },
+        };
+    }
+
+    Resource::LocString SettingsSetCommand::ShortDescription() const
+    {
+        return { Resource::String::SettingsSetCommandShortDescription };
+    }
+
+    Resource::LocString SettingsSetCommand::LongDescription() const
+    {
+        return { Resource::String::SettingsSetCommandLongDescription };
+    }
+
+    Utility::LocIndView SettingsSetCommand::HelpLink() const
+    {
+        return s_SettingsCommand_HelpLink;
+    }
+
+    void SettingsSetCommand::ValidateArgumentsInternal(Execution::Args& execArgs) const
+    {
+        // Get admin setting string for all available options except Unknown
+        std::vector<Utility::LocIndString> adminSettingList;
+        for (auto setting : GetAllSequentialEnumValues(StringAdminSetting::Unknown))
+        {
+            adminSettingList.emplace_back(AdminSettingToString(setting));
+        }
+
+        Utility::LocIndString validOptions = Join(", "_liv, adminSettingList);
+
+        if (execArgs.Contains(Execution::Args::Type::SettingName) && StringAdminSetting::Unknown == StringToStringAdminSetting(execArgs.GetArg(Execution::Args::Type::SettingName)))

```

Comment:
```suggestion
        if (StringAdminSetting::Unknown == StringToStringAdminSetting(execArgs.GetArg(Execution::Args::Type::SettingName))

```

nit: the SettingName arg is a required arg. Does not seem necessary to check again here.